cfpb/cfgov-refresh
# Careers 2.0: Add formatting options for dates

Adds formatting option for dates when they are displayed as text rather than headers (recommended by user testing on Careers pages & currently only applied in that section.) This PR is based on @anselmbradford's work in #3240 & review suggestions

## Additions

- strftime wrapper to allow additional patterns (custom month abbreviations, day values without leading zeros)
- date filter that adds option of formatting dates for text contexts in addition to the default header context

## Testing

1. Check out branch
2. Navigate to [careers landing page](http://localhost:8000/about-us/careers/), [current openings page](http://localhost:8000/about-us/careers/current-openings/), and a job opening page and verify the date formatting
3. Verify that other instances of dates on the site (like on [the Blog page](http://localhost:8000/about-us/blog/)) retain their existing styles 

## Screenshots

### Careers landing page
<img width="1140" alt="screen shot 2017-12-14 at 11 40 16 am" src="https://user-images.githubusercontent.com/778171/34020303-61cb7122-e0e8-11e7-9f81-5329f04966ae.png">

### Current openings page
<img width="1142" alt="screen shot 2017-12-14 at 11 40 03 am" src="https://user-images.githubusercontent.com/778171/34020305-6349fe42-e0e8-11e7-98e8-e8cdba047645.png">

### Job opening page
<img width="1145" alt="screen shot 2017-12-14 at 11 39 52 am" src="https://user-images.githubusercontent.com/778171/34020308-655084ae-e0e8-11e7-9387-c00524b8253f.png">

### Existing date format
<img width="774" alt="screen shot 2017-12-14 at 2 05 00 pm" src="https://user-images.githubusercontent.com/778171/34020340-8b182dc2-e0e8-11e7-839d-75b85e9b9585.png">


## Checklist
